lib/egl/egl_main.cpp

```diff
@@ -437,27 +424,24 @@ EglMainThread::Handle EglMainThread::CreateWindow(EglWindow *caller, int w,
    }
 #endif //GLVIS_USE_EGL
 #ifdef GLVIS_USE_CGL
-   if (!out_hnd.isInitialized())
-   {
-      return out_hnd;
+   if (!out_hnd.isInitialized()) { return out_hnd; }
+   CGLError err = CGLSetCurrentContext(out_hnd.ctx.get());
+   if (err != kCGLNoError) { return out_hnd; }
+   // initialize GLEW before using any OpenGL functions
+   const auto status = caller->initGLEW(legacy_gl);
+   MFEM_VERIFY(status, "GLEW initialization failed!");
+   assert(glGetError() == GL_NO_ERROR);
+   const auto *renderer = glGetString(GL_RENDERER);
+   if (strstr((const char*)renderer, "Software") != nullptr)
+   {
+      PRINT_DEBUG("Using software renderer; expect limited functionality");
    }
-
-   CGLError err = CGLSetCurrentContext(out_hnd.ctx);
-   if (err != kCGLNoError)
-   {
-      std::cerr << "Cannot set the CGL context as current, error: "
-                << CGLErrorString(err) << std::endl;
-
-      return out_hnd;
-   }
-
    glGenFramebuffers(1, &out_hnd.buf_frame);
+   glCheckFramebufferStatus(GL_FRAMEBUFFER);
```

Comment:
Is this good for something? The return value is not checked and the frame buffer is incomplete at this point anyway, because it does not have the render buffers attached yet.